app/code/core/Mage/Catalog/Model/Category/Api.php

```diff
@@ -464,7 +467,7 @@ public function assignProduct($categoryId, $productId, $position = null, $identi
     {
         $category = $this->_initCategory($categoryId);
         $positions = $category->getProductsPosition();
-        $productId = $this->_getProductId($productId);
+        $productId = $this->_getProductId($productId, $identifierType);
```

Comment:
Can you fix the docblock for `$identifierType` at line 462?

```
// from
* @param null $identifierType
// to
* @param null|string $identifierType Should be 'sku' when working with SKU's. Leave null when using ID's
```

Reply:
Added 😃